src/pages/index.js

```diff
@@ -103,22 +147,92 @@ function handleOverlay(event) {
 }
 
 function handleEditFormSubmit(evt) {
+    const editFormSubmitBtn = evt.submitter;
+    setButtonText(editFormSubmitBtn, true);
+
     evt.preventDefault();
-    profileName.textContent = editProfileName.value;
-    profileDescription.textContent = editProfileDescription.value;
-    closeModal(editProfileModal);
+    api.editUserInfo({ name: editProfileName.value, about: editProfileDescription.value })
+        .then((data) => {
+            profileName.textContent = data.name;
+            profileDescription.textContent = data.about;
+            closeModal(editProfileModal);
+        })
+        .catch(console.error)
+        .finally(() => {
+            setButtonText(editFormSubmitBtn, false);
+        });
 }
 
+function renderCard(item) {
+    const cardEl = getCardElement(item);
+    cardsList.prepend(cardEl);
+}
+
+
 function handleAddCardSubmit(evt) {
+    const addCardSubmitBtn = evt.submitter;
+    setButtonText(addCardSubmitBtn, true);
+
     evt.preventDefault();
-    const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-    const cardEl = getCardElement(inputValues);
-    cardsList.prepend(cardEl);
-    evt.target.reset();
-    disableButton(cardSubmitButton, settings);
-    closeModal(addCardModal);
+    api.addCard({ name: cardNameInput.value, link: cardLinkInput.value })
+        .then((cardData) => {
+            renderCard(cardData);
+            disableButton(cardSubmitButton, settings);
```

Comment:
*  You should **clear the inputs** after a successful adding of a new card to let the user add the 2nd card again without having to remove the old data manually.

Reply:
**Could be improved**


You can easily clear the inputs with `reset`. Every **form** has this method. And you have the form in each submit handler as `evt.target`. So, you can reset the form like this: `evt.target.reset()`

Or `cardForm.reset()`